daml-foundations/daml-tools/daml-cli/DA/Cli/Visual.hs

```diff
@@ -54,15 +55,18 @@ startFromExpr seen world e = case e of
 startFromChoice :: LF.World -> LF.TemplateChoice -> Set.Set Action
 startFromChoice world chc = startFromExpr Set.empty world (LF.chcUpdate chc)
 
-templatePossibleUpdates :: LF.World -> LF.Template -> Set.Set Action
-templatePossibleUpdates world tpl = Set.unions $ map (startFromChoice world) (NM.toList (LF.tplChoices tpl))
+data ChoiceAndAction = ChoiceAndAction { choiceForTemplate :: LF.Template , choice :: LF.TemplateChoice ,actions :: Set.Set Action }
```

Comment:
Do we need the `choiceForTemplate` field here? Isn't the `template` field below enough?

Reply:
Probably not. I added it for convenience